daml-foundations/daml-tools/da-hs-daml-cli/DA/Cli/Damlc.hs

```diff
@@ -208,7 +222,152 @@ execCompile inputFile outputFile opts = withProjectRoot $ \relativize -> do
+                            LF.versionDefault
+                            exposedModules
+                            dependencies
+                Managed.with (Compiler.newIdeState opts Nothing loggerH) $ \compilerH -> do
+                    darOrErr <-
+                        runExceptT $
+                        Compiler.buildDar
+                            compilerH
+                            main
+                            name
+                            [confFile]
+                            (UseDalf False)
+                    case darOrErr of
+                        Left errs ->
+                            ioError $
+                            userError $
+                            unlines
+                                [ "Creation of DAR file failed:"
+                                , T.unpack $
+                                  Pretty.renderColored $
+                                  Pretty.vcat $
+                                  map prettyDiagnostic $
+                                  Set.toList $ Set.fromList errs
+                                ]
+                        Right dar -> do
+                            let fp = targetFilePath name
+                            createDirectoryIfMissing True $ takeDirectory fp
+                            B.writeFile fp dar
+                            putStrLn $ "Created " <> fp <> "."
+  where
+    parseProjectConfig project = do
+        name <- queryProjectConfigRequired ["project", "name"] project
+        main <- queryProjectConfigRequired ["project", "source"] project
+        exposedModules <-
+            queryProjectConfigRequired ["project", "exposed-modules"] project
+        version <- queryProjectConfigRequired ["project", "version"] project
+        dependencies <-
+            queryProjectConfigRequired ["project", "dependencies"] project
+        Right (name, main, exposedModules, version, dependencies)
+
+    mkConfFile ::
+           String
+        -> String
+        -> LF.Version
+        -> [String]
+        -> [FilePath]
+        -> (String, B.ByteString)
+    mkConfFile name version lfVersion exposedMods deps = (confName, bs)
+      where
+        confName = name ++ ".conf"
+        lfVersionStr = lfVersionString lfVersion
+        bs =
+            BSC.pack $
+            unlines
+                [ "name: " ++ name
+                , "id: " ++ name
+                , "key: " ++ name
+                , "version: " ++ version
+                , "exposed: True"
+                , "exposed-modules: " ++ unwords exposedMods
+                , "import-dirs: ${pkgroot}" </> lfVersionStr </> name
+                , "library-dirs: ${pkgroot}" </> lfVersionStr </> name
+                , "data-dir: ${pkgroot}" </> lfVersionStr </> name
+                , "depends: " ++
+                  unwords [dropExtension $ takeFileName dep | dep <- deps]
+                ]
+
+    -- The default output filename is based on Maven coordinates if
```

Comment:
So I guess this means we’ll use Maven to distribute packages initially?

Reply:
This is from the old packaging command, I would not change it for now.